src/planet-dump.cpp

```diff
@@ -135,27 +145,63 @@ int main(int argc, char *argv[]) {
     std::vector<boost::shared_ptr<output_writer> > writers;
     if (options.count("history-xml")) {
       std::string output_file = options["history-xml"].as<std::string>();
-      writers.push_back(boost::shared_ptr<output_writer>(new xml_writer(output_file, options, display_name_map, max_time, true)));
+      writers.push_back(boost::shared_ptr<output_writer>(new xml_writer(output_file, options, 
+        display_name_map, max_time, /* clean */ false, /* history */ true)));
```

Comment:
I think the need to start commenting which booleans are which indicates that we should instead be using `enum class` constants for these. The advantage is two-fold; the values are self-describing, and they are type-safe and cannot be mixed up (as mentioned in a previous comment). For example, something like:

```c++
enum class user_info_level {
  FULL,
  ANON
};

enum class historical_versions {
  NONE,
  FULL
};
```

Then this invocation can be written as:

```c++
writers.push_back(boost::shared_ptr<output_writer>(new xml_writer(output_file, options, 
  display_name_map, max_time, user_info_level::FULL, historical_versions::FULL)));
```

What do you think?